source/user/mcscf.rst

```diff
@@ -1,57 +1,299 @@
-argument::
+CASSCF
+""""""
+.. code-block:: python
 
-  >>> mc = mcscf.CASSCF(m, 4, (4,2))
-  >>> print('E(CASSCF) = %.9g' % mc.kernel()[0])
-  E(CASSCF) = -149.609461
-  >>> print('S^2 = %.7f, 2S+1 = %.7f' % mcscf.spin_square(mc))
-  S^2 = 2.0000000, 2S+1 = 3.0000000
+  import pyscf
+  mol = pyscf.M(
+      atom = 'O 0 0 0; O 0 0 1.2',
+      basis = 'ccpvdz',
+      spin = 2)
+  myhf = mol.RHF().run()
+  ncas, nelecas = (6,8)
+  mycas = myhf.CASSCF(ncas, nelecas).run()
 
-The two integers in the tuple represent the number of alpha and beta electrons.
-Although it is a triplet state, the solution might not be correct since the
-CASSCF is based on the incorrect singlet HF ground state.  Starting from the
-ROHF ground state, we have::
+Like many other modules in PySCF, :mod:`mcscf` works with density-fitting (:numref:`user_df`) and x2c (:numref:`user_x2c`).
+It can also be used in its unrestricted formulation, please see :source:`examples/mcscf/60-uhf_based_ucasscf.py` for an example.
+An important feature of :mod:`mcscf` is that it can interface with external CI solver such as DMRG, FCIQMC, or selected CI methods, see the external projects for more details.
 
-  >>> mc = mcscf.CASSCF(rhf3, 4, 6)
-  >>> print('E(CASSCF) = %.9g' % mc.kernel()[0])
-  E(CASSCF) = -149.646746
 
-The energy is lower than the RHF initial guess.
-.. We can also use the UHF ground
-.. state to start a CASSCF calculation::
-.. 
-..   >>> mc = mcscf.CASSCF(uhf3, 4, 6)
-..   >>> print('E(CASSCF) = %.9g' % mc.kernel()[0])
-..   E(CASSCF) = -149.661324
-..   >>> print('S^2 = %.7f, 2S+1 = %.7f' % mcscf.spin_square(mc))
-..   S^2 = 3.9713105, 2S+1 = 4.1091656
-.. 
-.. Woo, the total energy is even lower.  But the spin is contaminated.
+
+Picking an Active Space
+-----------------------
+In general, selecting an active space can be cumbersome and PySCF offers severals ways to facilitate this process.
+There are four strategies and all require that you specify the number of electrons and orbitals in the active space.
+
+.. warning::
+  We always advise users to visualize their chosen active orbitals before starting large/expensive calculations.
+  This involves dumping the MO coefficients to a ``molden`` file (see example :source:`examples/tools/02-molden.py`) and visualizing with your chosen program.
+  While there are many great softwares available to visualize orbitals, `JMol <http://jmol.sourceforge.net/>`_ is one of the easiest to use and is recommended for less experienced users.
+
+
+1) (Default) Specifying no additional information.
+  This is the most minimal strategy for selecting an active space and chooses orbitals (and electrons) around the fermi-level that match the number of orbitals and electrons specified by the user.
+  In most circumstances, this is not an ideal strategy and will lead to poor convergence or none at all.
+  For example:
+.. code-block:: python
+
+  ncas, nelecas = (6,8)
+  mycas = myhf.CASSCF(ncas, nelecas)
+
```

Comment:
The traditional ways to pick active spaces are missing here. Often one runs an MP2 or CISD natural orbital calculation; and then picks the active space based on these natural orbitals. This leads to a much better guess than just HF orbitals.

Several other ways to pick the active space have also been suggested such as improved virtual orbitals (IVOs), which should be pretty easy to form in PySCF.

Reply:
You're right that I missed the natural orbitals, I can add those here. I was focusing strictly on methods where PySCF already contained the convenience functions for them in the mcscf modules. The natural orbitals are easy enough though so I can show that in a few lines.

Reply:
Yes, please add an example to PySCF and `include` it here.

Reply:
There is also the widely used strategy is to use just the minimal active that consists of either singly occupied orbitals or all transition-metal d orbitals.

Reply:
@bogdanoff that's already covered by the AVAS example

However, what is also missing is Pulay's unrestricted HF natural orbitals, which should likewise be easy to implement in PySCF.